facilitator/src/http.rs

```diff
@@ -1,65 +1,127 @@
-use anyhow::{anyhow, Context, Result};
-use ureq::{Request, Response, SerdeValue};
+use anyhow::{Context, Result};
+use std::fmt::Debug;
+use std::time::Duration;
+use ureq::{Agent, AgentBuilder, Request, Response, SerdeValue};
+use url::Url;
 
-use crate::gcp_oauth::OauthTokenProvider;
+#[derive(Debug)]
+pub(crate) enum Method {
+    Get,
+    Post,
+    Put,
+    Delete,
+}
+
+impl Method {
+    fn to_primitive_string(&self) -> &str {
+        match self {
+            Method::Get => "GET",
+            Method::Post => "POST",
+            Method::Put => "PUT",
+            Method::Delete => "DELETE",
+        }
+    }
+}
+
+fn create_agent() -> Agent {
+    AgentBuilder::new().timeout(Duration::from_secs(10)).build()
+}
+
+pub(crate) fn prepare_request(
+    agent: Option<Agent>,
+    parameters: RequestParameters<'_>,
+) -> Result<Request> {
+    let agent = agent.unwrap_or(create_agent());
+
+    let request = agent.request_url(parameters.method.to_primitive_string(), &parameters.url);
+    parameters.authenticated_request(request)
+}
+
+pub(crate) trait OauthTokenProvider: Debug {
+    fn ensure_oauth_token(&mut self) -> Result<String>;
+}
+
+#[derive(Debug)]
+pub(crate) struct StaticOauthTokenProvider {
+    pub token: String,
+}
+
+impl OauthTokenProvider for StaticOauthTokenProvider {
+    fn ensure_oauth_token(&mut self) -> Result<String> {
+        Ok(self.token.clone())
+    }
+}
+
+impl std::convert::From<String> for StaticOauthTokenProvider {
+    fn from(token: String) -> Self {
+        StaticOauthTokenProvider { token }
+    }
+}
 
 /// Struct containing parameters for send_json_request
```

Comment:
This comment is no longer accurate since I don't pass `RequestParameters` to `send_json_request`.